### ISSUE TITLE
Display connected to Dell WD15 dock won't work

### ISSUE THREAD

**Issue by ozgunawesome** [kfix/ddcctl#50]
I have one U3415W and one U3417W display running on my MacBook Pro (13-inch, 2017, Two Thunderbolt 3 ports) running macOS Sierra 10.12.6 (16G1710)

One of the displays are connected through a Dell WD15 docking station via the HDMI port. The other goes in over HDMI through an Apple USB Type-C Digital AV Adapter. 

Whatever I do, I can't control the one display connected through the docking station. I get the following:

```
$ ./ddcctl -d 2 -b 1
D: NSScreen #724048205 (2560x1080 0°) 72.00 DPI
D: NSScreen #724072846 (2560x1080 0°) 72.00 DPI
I: found 2 external displays
I: polling display 2's EDID
I: got edid.serial: FR3PK7BG03NL
I: got edid.name: DELL U3417W
D: action: b: 1
D: setting VCP control #16 => 1
E: Failed to send DDC command!
```

It doesn't matter if it's this display or the other. I can't control the one that's hooked up to the dock. If I hook it through the USB-C adapter I can control it just fine.

I'd really like to have this feature so I want to do whatever I can to fix this. So if you need more debug info, I can provide. Meanwhile I'm fiddling with the source myself. What could be the cause of this?

**Comment by am0** [comment]
Having the same issue when hooking my monitor up through an adaptor- one of these: [https://satechi.net/products/aluminum-type-c-multimedia-adapter](https://satechi.net/products/aluminum-type-c-multimedia-adapter). Previously it was connected via a USB-C to Mini Displayport adaptor and it worked fine, but now I get the same 'Failed to send DDC command!' error.

Assume something about it is blocking commands. Wouldn't mind helping to fix this if I can provide any further details!

**Comment by JoniVR** [comment]
Yup, connectors matters. Some connectors require a different `minReplyDelay` and some won't even work at all using a specific connector.

**Comment by heyoni** [comment]
Do we know if ddcctl works with any docks, or is it all docks?

**Comment by theogq** [comment]
Tested ddcctl on my 2018 macbook (pro running Mojave 10.14.5) that is connected to two benq BL2410 monitors (both monitors are connected using DP-cables) through ThinkPad Hybrid USB-C with USB-A Dock, and ddcctl doesn't work either.

`ddcctl -d 1 -i \?
2019-10-21 13:15:53.232 ddcctl[57180:1400025] D: NSScreen #1952486408 (1920x1080 0°) 102.00 DPI
2019-10-21 13:15:53.232 ddcctl[57180:1400025] D: NSScreen #1952486407 (1920x1080 0°) 102.00 DPI
2019-10-21 13:15:53.232 ddcctl[57180:1400025] I: found 2 external displays
2019-10-21 13:15:53.232 ddcctl[57180:1400025] I: polling display 1's EDID
2019-10-21 13:15:53.242 ddcctl[57180:1400025] E: Failed to poll display!`

**Comment by xdsoar** [comment]
> Do we know if ddcctl works with any docks, or is it all docks?

In my case (macbook pro 13-inch 2017 os version 10.14.6). 
ddcctl works well with belkin thunderbolt3 dock.
I also test with thinkpad thunderbolt dock, which doesn't work.

**Comment by JBFUK** [comment]
Dell D6000 dock, 3x S2721DGF monitors.  I can only adjust the monitor connected to the HDMI port (1).  The two monitors connected to the DisplayPorts (2/3) give the same error as above.

D: NSScreen #722486414 (2560x1440 0°) 109.00 DPI
D: NSScreen #722486548 (2560x1440 0°) 109.00 DPI
D: NSScreen #722489109 (2560x1440 0°) 109.00 DPI
I: found 3 external displays
I: polling display 2's EDID
E: Failed to poll display!

**Comment by cfiske** [comment]
It doesn't work for me with a Dell WD19TB Thunderbolt dock, display (Dell U3419W) connected via dock's HDMI port. The EDID part works fine and it correctly identifies the monitor, just can't send commands. Same output/error as OP. It works fine connecting directly via USB-C.

**Comment by ozgunawesome** [comment]
@cfiske  I think it has something to do with the dock. The way I saw it, the dock must be able to pass through DDC commands to the display, and my dock probably didn't have that feature.

Adapters such as the one you linked (and the original Apple one as well) usually pass the USB-C port's monitor lines directly to the monitor so they usually work fine. It's the Thunderbolt/USB-C docks with multiple display inputs that have this issue.

**Comment by AngryKiller** [comment]
I have the same issue with my Dell WD15, but DDC commands work when I boot into Windows, so the issue may not be impossible to resolve.
